### DISCUSSION TITLE
[Feature] Download/Upload sidecar metadata

### DISCUSSION THREAD

**Discussion by ayykamp** [immich-app/immich#10592] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

When downloading an image it should be possible to also download the sidecar xmp files. Especially also when downloading an entire album, it would be nice to have an option to also downloads the sidecar files.

Additionally, when uploading assets through the web ui it should be possible to also upload sidecar files which then get associated with the assets by file name.

Thank you !

### Platform

- [ ] Server
- [X] Web
- [ ] Mobile

**Comment by Marty56** [comment]
This is an essential feature. 
Just keeping the meta data within some obscure database is a showstopper.
Basically all the work a user has put in updating its meta data within immich is lost, when he wants to move to a different photo tool.

**Reply by ayykamp** [reply]
That's not true. The xmp files are present in the file system, this is just about the web ui.

**Comment by clamel77** [comment]
I agree with this feature request, especially downloading albums

**Comment by patrontheo** [comment]
Hey ! 
I am backing this FR with my use-case here:

I am switching to a new phone, and would like to have my recent photo library available offline (let's say the photos this year).
I have tried to download them from the immich ios app and they appear in the photo library and are recognized by immich as being already uploaded so all good on that side.
The only issue is that some of my photos had no date / location when i imported them in immich and I manually set the date / location in immich. So this metadata is written in the XMP sidecar. 
BUT, when I download from immich ios app, it seems it only download the photo and not the xmp sidecar in the camera roll, so all the photos without date appear as today in my library (ios app, not immich app).
This is not horrible but I'm wondering if it would be possible that immich rewrites the ios xmp sidecar file so that the ios library app reads everything correctly. Or is this technically undoable ?

Thankksss

**Reply by fabb** [reply]
Maybe there could be 2 download buttons `Download Original` and `Download with updated EXIF data`. The second one could on-the-fly integrate the sidecar data into the image file itself, just for the downloaded file.

**Comment by JoeHardi** [comment]
Any news on that?

**Reply by bo0tzz** [reply]
https://justinmayer.com/posts/any-updates/

**Reply by JoeHardi** [reply]
I guess thats one on me

**Comment by JoeHardi** [comment]
Then let me at least add something I have recognized:

When uploading pictures where the location was already set up, the location appears in the downloads.

When uploading pictures without location and setting up the location within immich, the location will not appear in the aforementioned downloaded pictures.

**Reply by KittKat7** [reply]
My understanding is that editing metadata in Immich does not affect the original file, but instead sets that metadata in the database/a sidecar file. When you download the image, you only get the original image (without the sidecar file or the database's data) so it only has the metadata present when you uploaded it originally.

**Comment by KittKat7** [comment]
I would love to see this (specifically downloading) added as a feature. I often need to adjust my pictures metadata after I take the picture. If I share these images with my friends, I would like the metadata updates to be included. I understand that Immich does not want to take any possibly destructive action with the original file.

I think a good solution would be to have the `Download` button create and download a temporary copy of the image (with the metadata merged in), but leaving the original file untouched on the system. For those who want the originally uploaded file, add a `Download Original` button that downloads the original file as stored on the system. This way its possible to get the file with the updated metadata from the UI, while not modifying the original file.
